### ISSUE TITLE
Forms: Required field legend

### ISSUE THREAD

**Issue by humancompanion** [USAJOBS/design-system#23]
We need to design and create guidelines around a required field legend. We don't have guidance from the Standards as of yet on this and it will be in My Account as well as other sections of the site.

#### Tasks
* [x] Define whether or not the legend is in fact required on every screen at every viewport
* [x] Define location of the "Required" legend
* [x] Define visual style of "Required" legend.

My sense is that the legend should be somewhere. There may be guidance in the 508 standards around this, but I couldn't locate them easily.

**Comment by humancompanion** [comment]
An example of the current state:

<img width="1193" alt="screen shot 2015-11-06 at 8 37 13 am" src="https://cloud.githubusercontent.com/assets/352987/11131101/cf0f28be-8956-11e5-8b3d-ba0d8b3f43d4.png">
